test/wordsearch.js

```diff
@@ -0,0 +1,79 @@
+/* global describe, it */
+
+const assert = require('assert');
+const { WordSearchSolver } = require('../');
+
+describe('WordSearchSolver', function () {
+  it('Cardinal directions', function () {
+    const matrix = [
+      [ 'p', 'x', 'x', 'x' ],
+      [ 'u', 'w', 'i', 'n' ],
+      [ 'z', 'x', 'x', 'x' ],
+      [ 'z', 'x', 'x', 'x' ]
+    ];
+    const solver = new WordSearchSolver(matrix);
+    const results = solver.findWords(['puzz', 'win', 'foo', 'bar', 'baz']);
+
+    assert.strictEqual(results.length, 2);
+  });
+
+  it('Diagonal directions', function () {
+    const matrix = [
+      [ 'p', 'x', 'x', 'x', 'b' ],
+      [ 'x', 'u', 'x', 'a', 'x' ],
+      [ 'x', 'x', 'z', 'x', 'x' ],
+      [ 'x', 'x', 'x', 'z', 'x' ]
+    ];
+    const solver = new WordSearchSolver(matrix);
+    const results = solver.findWords(['puzz', 'baz']);
+
+    assert.strictEqual(results.length, 2);
+  });
+
+  it('Jagged array outside box', function () {
+    const matrix = [
+      [ 'x', 'f', 'o', 'o' ],
+      [ 'b', 'x', 'x', ],
+      [ 'a', 'x', 'b', 'a', 'r' ],

```

Comment:
Can you have this one go backwards to ensure that we're checking for words starting outside the non-jagged part of the array?

Reply:
I think the next test covers this case?